itenium-socks/src/app/socks/shop.component.ts

```diff
@@ -1,21 +1,44 @@
-import { Component } from '@angular/core';
+import {Component, OnInit} from '@angular/core';
 import { SocksService } from './socks.service';
-import { Observable } from 'rxjs';
+import {last, Observable} from 'rxjs';
 import { Sock } from './sock.model';
 import { AsyncPipe, NgFor } from '@angular/common';
+import {MatPaginator, PageEvent} from "@angular/material/paginator";
 
 @Component({
   selector: 'app-shop',
   standalone: true,
-  imports: [NgFor, AsyncPipe],
+  imports: [NgFor, AsyncPipe, MatPaginator],
   templateUrl: './shop.component.html'
 })
-export class ShopComponent {
+export class ShopComponent implements OnInit{
+  totalItems = 100;
+  pageSize = 8;
+  currentPage = 0;
```

Comment:
In een prod applicatie zou ik hier ook een apart component van maken

```
class Paginator {
  @Output() pageChanged = new EventEmitter<PaginationInfo>();
}
```

Je wil paginatie toevoegen aan de SocksShop, maar later ook op de ReviewsPage en op de UsersPage en op de ...
Maar je wil de paginatie code niet elke keer gaan herhalen!

vb:
```
class PagedItemComponent {}
```

Dit is al iets ingewikkelder, je zal vb met een slot moeten werken: `<ng-content></ng-content>`

Grote voordeel:
Je ShopComponent begaat geen SRP violations: nu gaat de ShopComponent paginatie code bevatten, en later ook Filter code, en sorteer code etc...

En je kan het hergebruiken voor alles waardoor je bugfixes (off by 1 errors?) en changes (de paginatie buttons moeten anders gepositioneed & gestyled worden) maar op 1 plaats moet doen